freebsd-stubs.h

```diff
@@ -0,0 +1,14 @@
+#ifndef _FREEBSD_STUBS_H
+#define _FREEBSD_STUBS_H
+
+#define O_PATH 0
+#define O_DSYNC 0
+#define O_LARGEFILE 0
+#define O_NOATIME 0
+#define O_TMPFILE 0
+#define AT_EMPTY_PATH 0
```

Comment:
The `AT_EMPTY_PATH` thing wouldn't work like that, really on FreeBSD the `fstatat` calls with this arg should be changed to regular old `fstat`.

IIUC, the point of this is that `fstatat(fd, "", &thing, AT_EMPTY_PATH|AT_SYMLINK_NOFOLLOW)` means "do fstat on the `O_PATH` symlink file descriptor itself" and we don't support `O_PATH`. So `fstatat` with the empty path would blow up on FreeBSD.